### ISSUE TITLE
Proposal: _argv calls should always require user_context

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#768]
Having user_context be an optional argument (controlled by a Target feature flag) is a somewhat regrettable wart; the motivation for making it optional is to keep the default calling convention as simple as possible, since most users don't need a custom user_context.

However, it complicates some build environments, especially those that use the _argv() call, since the calling patterns now vary depending on how the filter was built (ie what Target flags were active).

Since the _argv() calls are already awkward and not-used-by-beginners, let's simplify that part of the situation by just declaring that the _argv() call *always* takes user_context as the first entry, regardless of Target::UserContext; runtimes that don't care about it can just set it to null.

**Comment by abadams** [comment]
If the argv exists, isn't it easy to check the metadata to see if the
function expects a user context?

**Comment by steven-johnson** [comment]
Yeah, I didn't explain myself well at all here. My motivation isn't to simplify the pattern at the call site; my motivation is to always have a calling API that allows me to pass a user_context, regardless of how the UserContext flag is set.

**Comment by abadams** [comment]
Oh yeah, that makes sense. Would the inner code reliably pass the
user_context through to the runtime though? if the argv version just calls
the regular version, there's nowhere to put the arg. It should be possible
to make this work, but it would take some changes to CodeGen_LLVM, right?

**Comment by steven-johnson** [comment]
It might take a little nooding, yeah, but not too much -- just an alternate
(private) entry point.

**Comment by abadams** [comment]
+1 then

**Comment by abadams** [comment]
Covered by #7298
